### ISSUE TITLE
A.7.6 Coupled resource

### ISSUE THREAD

**Issue by idevisser** [SEMICeu/GeoDCAT-AP#140]
GeoDCAT-AP :The target dataset or series SHOULD be preferably referred to by using its unique resource identifier

but 

TG Requirement 3.6: metadata/2.0/req/sds/coupled-resource

Links pointing to the online metadata descriptions of data sets provided by the described service shall be given using srv:operatesOn element.

The multiplicity of this element is 0..n.

This property shall be implemented by reference. The xlink:href attribute of each of the srv:operatesOn elements shall contain a URI pointing to the metadata record of the provided data set or data set series.

expecting an unique resource identifier is not realistic

**Comment by bertvannuffelen** [comment]
@idevisser can you elaborate your comment with an example?

As the linking is done by reference according to your message: there is an identifier. 
Is your comment on the uniqueness, the URI representation, or something else?

**Comment by idevisser** [comment]
The coupled resource contains the URL of the metadata describing the dataset, it does not contain the unique resource identifier as specified in https://github.com/INSPIRE-MIF/technical-guidelines/blob/main/metadata/metadata-iso19139/metadata-iso19139.adoc#3121-unique-resource-identifier. 
Although [ISO 19119:2016](https://www.iso.org/standard/59221.html?browse=tc) requires that the target of the <srv:operatesOn> element (Coupled Resource) is the #MD_DataIdentification class of the linked dataset metadata, due to the different existing metadata implementations, is concluded to allow a more simple implementation of the Coupled Resource linkage,  a URI pointing to the metadata record.

**Comment by hallinpihlatie** [comment]
True. It includes the metadata file identifier, which is highly likely to be unique, but its not the same as the unique resource identifier. 

An xml snippet of a service metadata, where the file identifier of the dataset metadata has been included, both as is and as a href link:

```xml
<srv:coupledResource>
    <srv:SV_CoupledResource>
      <srv:operationName>
        <gco:CharacterString>GetCapabilities</gco:CharacterString>
      </srv:operationName>
      <srv:identifier>
        <gco:CharacterString>562b0259-8e8f-447a-933a-aa27d57ef8d9</gco:CharacterString>
      </srv:identifier>
      <gco:ScopedName>undefined</gco:ScopedName>
    </srv:SV_CoupledResource>
  </srv:coupledResource>
  <srv:couplingType gco:nilReason="missing" />
  <srv:containsOperations gco:nilReason="missing" />
  <srv:operatesOn uuidref="562b0259-8e8f-447a-933a-aa27d57ef8d9" xlink:href="http://www.paikkatietohakemisto.fi/geonetwork/srv/eng/csw?service=CSW&amp;request=GetRecordById&amp;version=2.0.2&amp;outputSchema=http://www.isotc211.org/2005/gmd&amp;elementSetName=full&amp;id=562b0259-8e8f-447a-933a-aa27d57ef8d9" />
</srv:SV_ServiceIdentification>
```

**Comment by jakubklimek** [comment]
@idevisser @hallinpihlatie Could you elaborate on what your proposal is?

If I read it correctly, `srv:operatesOn/@xlink:href` is currently being interpreted as an IRI of of a dataset or dataset series a dataservice provides access to, mapped to `dcat:servesDataset` as oer [GeoDCAT-AP 3.0.0](https://semiceu.github.io/GeoDCAT-AP/releases/3.0.0/#:~:text=coupled%20resource%20(o)) section [A.5.1 Overview of bindings for GeoDCAT-AP Core](https://semiceu.github.io/GeoDCAT-AP/releases/3.0.0/#overview-of-bindings-for-geodcat-ap-core)

You are saying that this will not be an IRI of the dataset/dataset series and that instead, it will be a URL of the metadata record of that dataset/dataset series, which would have to be downloaded and processed in order to get its IRI?

This situation seems similar to the problem of data service metadata being removed from INSPIRE metadata in favor of dereference, which makes it then difficult to handle using XSLT mapping.

**Comment by idevisser** [comment]
The coupled resource is in the INSPIRE metadata be implemented by reference. The ISO example doesn't match this requirement. The example above, provided by Lena illustrates better the situation.

**Comment by jescriu** [comment]
> You are saying that this will not be an IRI of the dataset/dataset series and that instead, it will be a URL of the metadata record of that dataset/dataset series, which would have to be downloaded and processed in order to get its IRI?

In A.7.6 Coupled resource (in GeoDCAT-AP): The target dataset or series SHOULD be preferably referred to by using its unique resource identifier.

In ISO/INSPIRE service metadata, as per TG Requirement 3.6, we indirectly refer to the dataset (or dataset series) through the `coupled resource`: 
* It contains a URI pointing to the dataset metadata record (using the metadata identifier, i.e. `fileIdentifier`). 
* It shall be implemented by reference. The `xlink:href` attribute of each of the `srv:operatesOn` element shall contain this URI (pointing to the metadata record describing the dataset (or dataset series).
* In certain cases this URI contains an explicit pointer to the #MD_DataIdentification class of the metadata record - which represents and describes the linked dataset (or dataset series) - THIS is required by [ISO 19119:2016](https://www.iso.org/standard/59221.html?browse=tc). 

Regarding the last bullet, INSPIRE allows a more simple implementation of the Coupled Resource linkage (e.g. a URI pointing to the dataset or dataset series metadata record), reason why it is not available in all implementations.

Therefore, the Coupled Resource is not exactly (not always) a unique resource identifier of the dataset (or series).

This is why @idevisser said:

> expecting an unique resource identifier is not realistic

In view of discussing this issue in the upcoming [Kick-off meeting of Working Team T-1A](https://github.com/INSPIRE-MIF/GeoDCAT-AP-pilot/tree/main/meetings/2026-02-24%20-%20T-1A%20Kick-off%20) of the ISO & GeoDCAT-AP Pilot.

I do not think this is really an issue. GeoDCAT-AP says:

> The target dataset or series **SHOULD** be preferably referred to by using its unique resource identifier.

In my view, this SHOULD takes into account the relaxed implementation allowed in INSPIRE, and this could be explained in a usage note in GeoDCAT-AP.

**Comment by jescriu** [comment]
Dear @idevisser, @hallinpihlatie,

Would you be willing to draft the content of this usage note?

**Comment by jescriu** [comment]
Drafting the proposed usage note is pending. Proposals are welcomed.

**Comment by streino** [comment]
Mapping `gmd:parentIdentifier` will hit the same issue: `gmd:parentIdentifier` contains a *record/metadata ID* but the (draft) proposed mapping `dcat:inSeries` expects a *resource ID* ([more details here](https://github.com/SEMICeu/iso-19139-to-dcat-ap/issues/78#issuecomment-4487128507)).
